cumulus/parachains/runtimes/coretime/coretime-rococo/src/coretime.rs

```diff
@@ -205,26 +270,25 @@ impl CoretimeInterface for CoretimeAllocator {
 		}
 	}
 
-	fn check_notify_revenue_info() -> Option<(RCBlockNumberOf<Self>, Self::Balance)> {
-		let revenue = CoretimeRevenue::get();
-		CoretimeRevenue::set(&None);
-		revenue
+	fn check_notify_revenue_info(
+	) -> Option<OnDemandRevenueRecord<RCBlockNumberOf<Self>, Self::Balance>> {
+		RevenueInbox::<Runtime>::take()
 	}
 
 	#[cfg(feature = "runtime-benchmarks")]
-	fn ensure_notify_revenue_info(when: RCBlockNumberOf<Self>, revenue: Self::Balance) {
-		CoretimeRevenue::set(&Some((when, revenue)));
+	fn ensure_notify_revenue_info(
+		info: OnDemandRevenueRecord<RCBlockNumberOf<Self>, Self::Balance>,
+	) {
+		RevenueInbox::<Runtime>::put(info);
 	}
```

Comment:
This logic makes no sense. Why do we put this into the internal storage item and then require that the external trait implementation returns the internal storage value? Either this trait is implemented by some completely different pallet upstream or we remove these two methods.

Reply:
I think @seadanda had some reasoning why it should be kept like this :thinking: 
@seadanda ?

Reply:
Found it: https://github.com/paritytech/polkadot-sdk/pull/3940/commits/3decbf37eeae90de0278b575a48d38b62f98c32b

Reply:
He says it should not be tightly coupled, but what you are doing here is still the same as before. It is still tightly coupled. You basically "cheated" to fix his comment, but the underlying solution is still the same.

Reply:
Yes, exactly. He suggests we keep the methods where they are not to break the interface, and later, we move the `RevenueInbox` into the coretime allocator pallet when we have it, thus decoupling things and not introducing breaking changes to the public interface twice. I agree that the current implementation is kind of cheating, but it's for good.

Reply:
It is not for "good"

Reply:
So what do you suggest? Remove it from the interface completely?

Reply:
This entire code path makes no sense. Just remove the trait function and then it is good. You also added the dispatchable to report the revenue to the pallet. I really don't see any reasons for the gymnastics you are doing there.